frontend/src/component/changeRequest/ChangeRequest/Changes/Change/strategy-change-diff-calculation.ts

```diff
@@ -53,80 +58,48 @@ export const getChangesThatWouldBeOverwritten = (
-                        snapshotValue,
-                        currentValue,
-                        changeValue,
-                        fallback,
-                    });
+            const changeInfo = {
+                property: key as keyof ChangeRequestEditStrategy,
+                oldValue: currentValue,
+                newValue: changeValue,
+            };
 
-                // compare, assuming that order never changes
-                if (key === 'segments') {
-                    // segments can be undefined on the original
-                    // object, but that doesn't mean it has changed
-                    if (hasJsonDiffWithFallback([])) {
-                        return {
-                            property: key as keyof ChangeRequestEditStrategy,
-                            oldValue: currentValue,
-                            newValue: changeValue,
-                        };
-                    }
-                } else if (key === 'variants') {
-                    // strategy variants might not be defined, so use
-                    // fallback values
-                    if (hasJsonDiffWithFallback([])) {
-                        return {
-                            property: key as keyof ChangeRequestEditStrategy,
-                            oldValue: currentValue,
-                            newValue: changeValue,
-                        };
-                    }
-                } else if (key === 'title') {
-                    // the title can be defined as `null` or
-                    // `undefined`, so we fallback to an empty string
-                    if (hasJsonDiffWithFallback('')) {
-                        return {
-                            property: key as keyof ChangeRequestEditStrategy,
-                            oldValue: currentValue,
-                            newValue: changeValue,
-                        };
-                    }
-                } else if (
-                    hasChanged(snapshotValue, currentValue, changeValue)
-                ) {
-                    return {
-                        property: key as keyof ChangeRequestEditStrategy,
-                        oldValue: currentValue,
-                        newValue: changeValue,
-                    };
-                }
-            })
-            .filter(
-                (
-                    change,
-                ): change is DataToOverwrite<keyof ChangeRequestEditStrategy> =>
-                    Boolean(change),
-            );
+            if (key in fallbacks) {
+                return hasJsonDiff({
+                    snapshotValue,
+                    currentValue,
+                    changeValue,
+                    fallback,
+                })
+                    ? changeInfo
+                    : undefined;
+            } else if (hasChanged(snapshotValue, currentValue, changeValue)) {
+                return changeInfo;
+            }
+        })
+        .filter(
+            (
```

Comment:
again I'd name this function to express the intent

Reply:
Yeah, that's fair. I'll fix.

Reply:
30d8a2cc3123780d7529b5fb1420dbda76e0b8a0